app/actions/local/post.ts

```diff
@@ -131,3 +131,26 @@ export const removePost = async (serverUrl: string, post: PostModel) => {
 export const selectAttachmentMenuAction = (serverUrl: string, postId: string, actionId: string, selectedOption: string) => {
     return postActionWithCookie(serverUrl, postId, actionId, '', selectedOption);
 };
+
+export const markPostAsDeleted = async (serverUrl: string, post: Post) => {
+    const operator = DatabaseManager.serverDatabases[serverUrl]?.operator;
+    if (!operator) {
+        return {error: `${serverUrl} database not found`};
+    }
+
+    const dbPost = await queryPostById(operator.database, post.id);
+    if (!dbPost) {
+        return {};
+    }
+
+    dbPost.prepareUpdate((p) => {
```

Comment:
the removal of everything that relates to the post is being removed with the action `removePost`

Reply:
Regarding cleanup, sure, removePost will take care of it. But here my concern is about sanitation. Consider the following scenario:

User A sends a message with a file attached that should not be attached.
User B receives the message, with the needed fileInfos.
User A deletes the message.
User B receives the websocket event, and the post is updated so it is visible as a deleted post.
User B uses some tool to check the database, and take out the fileInfos.
User B is able to reference (or see information about) the files in the message.

Reply:
> Regarding cleanup, sure, removePost will take care of it. But here my concern is about sanitation. Consider the following scenario:
> 
> 
> 
> User A sends a message with a file attached that should not be attached.
> 
> User B receives the message, with the needed fileInfos.
> 
> User A deletes the message.
> 
> User B receives the websocket event, and the post is updated so it is visible as a deleted post.
> 
> User B uses some tool to check the database, and take out the fileInfos.
> 
> User B is able to reference (or see information about) the files in the message.

I totally follow, but let's say user B uses a tool to inspect the DB and grt the fileInfo, this user will need to retrive the file from the server, in which it ahould be the server denying access, shouldn't it?

I do wonder if we need to remove a file that has already been downloaded though, which may be a bit tricky for images

Reply:
I guess you are right. Taking a look to the FileInfo type, the only thing that may still "worry" me a bit, is the name of the file (it may be something we don't want the user to read if the message has been deleted).

Regarding the deletion of the local file, seems tricky. Not sure if needed.